### PR TITLE
[CSL-1977] Review of US doc: fixes

### PR DESCRIPTION
**Pull request by georgeee**

### CODE REVIEW

File: docs/block-processing/us.md [hunk 1]
```diff
@@ -152,23 +168,27 @@ So if a value is going to be updated then a field is `Just` and `Nothing` otherw
 
 ### Adoption of block version
 
-Informally, `BlockVersion` is **adopted** if sum of block issuers' stakes, 
-which issued blocks of this `BlockVersion` at least once, takes a significant part of the stake.
+Informally, `BlockVersion = (Maj, Min, Alt)` is **adopted** iff:
```
Comment (georgeee) [inline docs/block-processing/us.md]: This and following three sections should be unified into one and rewritten in consistent way

**Comment by pva701** [comment]
First of all: the arrow in the picture `docs/block-processing/us-bv-states.png` with label "When other competing becomes Adopted" seems to be wrong because if "other version" is less then the competing one, there is a chance that competing versions becomes Adopted

File: docs/block-processing/us.md [hunk 2]
```diff
@@ -148,55 +218,50 @@ data BlockVersionModifier = BlockVersionModifier
...
-Also the following rules regarding alternative versions take place (assuming checks above passed):
-  * If `(Major, Minor)` of proposed version is equal to `(Major, Minor)` of
-    last adopted version, then alternative version must be equal to alternative version of last adopted version.
-  * Otherwise `(Major, Minor)` of proposed version is lexicographically greater 
-    than `(Major, Minor)` of last adopted version. In this case
-    other proposed block versions with same `(Major, Minor)` are considered. 
-    Let's call set of these block version `X`:
-    * If `X` is empty, given alternative version must be `0`.
-    * Otherwise it must be in `X` or greater than maximum from `X` by `1`.
+When some block version becomes *adopted*, all other *competing* block versions are never to become *adopted*.
```
Comment (pva701) [inline docs/block-processing/us.md]: as I said before, it's wrong.

File: docs/block-processing/us.md [hunk 3]
```diff
@@ -148,55 +218,50 @@ data BlockVersionModifier = BlockVersionModifier
...
+
+Note, that at any point in time all *competing* block versions have same major and minor versions (i.e. differ only in alt version).
```
Comment (pva701) [inline docs/block-processing/us.md]: this is also wrong. we can have such set of competing versions `{1.0.1, 1.1.0}`, when `adopted = 1.0.0`

Comment (pva701) [reply docs/block-processing/us.md]: so it clarifies why your previous sentence 
>When some block version becomes *adopted*, all other *competing* block versions are never to become *adopted*.

is wrong: we can adopt the former and then the latter,

File: docs/block-processing/us.md [hunk 4]
```diff
@@ -35,28 +35,62 @@
...
 ```
 data UpdatePayload = UpdatePayload
     { upProposal :: !(Maybe UpdateProposal)
     , upVotes    :: ![UpdateVote]
     }
 ```
 
-This document describes update system model and consensus rules including update payload verification, 
-which is part of block payload verification. It's described formally [here](overall.md#task-definition).
+#### TODO: Update proposal
```
[the remark this answers is not in the document]
Comment (pva701) [reply docs/block-processing/us.md]: ok, we decided to insert here bullet list with refers to the next sections

File: docs/block-processing/us.md [hunk 5]
```diff
@@ -72,49 +106,66 @@ and it has more stakes "for" than "against", then the proposal becomes approved
...
+
 _Approved_ proposal may become _active_ or even _rejected_ if rollback occurs.
 
 * **Rejected**  
-If a decision about proposal has been made negatively, then proposal become **rejected**.
-_Rejected_ proposal may become _active_ or even _approved_ if rollback occurs.
+If a decision about proposal has been made negatively, the proposal becomes **rejected**.
+
+_Rejected_ proposal may become _active_ or even _approved_ if block rollback occurs
+(and poll ended with positive decision on alternative branch).
```
[the remark this answers is not in the document]
Comment (georgeee) [reply docs/block-processing/us.md]: Removed brackets

File: docs/block-processing/us.md [hunk 6]
```diff
@@ -72,49 +106,66 @@ and it has more stakes "for" than "against", then the proposal becomes approved
...
 
 * **Confirmed**  
-If a proposal has been _approved_ in some block and there are at least `k` blocks after this one, then
+If a proposal has been _approved_ in some block and there are at least `k` blocks after this one, the
```
[the remark this answers is not in the document]
Comment (georgeee) [reply docs/block-processing/us.md]: You don't have to use *then* here. You have implication already by using `If`.

Article *the* is syntactically correct form of *proposal* word usage

File: docs/block-processing/us.md [hunk 7]
```diff
@@ -72,49 +106,66 @@ and it has more stakes "for" than "against", then the proposal becomes approved
...
 update proposal becomes **confirmed**.
+
```
[the remark this answers is not in the document]
Comment (georgeee) [reply docs/block-processing/us.md]: ok

File: docs/block-processing/us.md [hunk 8]
```diff
@@ -72,49 +106,66 @@ and it has more stakes "for" than "against", then the proposal becomes approved
...
+Though the proposal is committed to the blockchain, 
+stakeholders are still to be decide whether to approve or to reject this proposal.
 So, a poll is active and stakeholders' votes which get into the blockchain affect the decision.
 
 * **Approved**  
-If a decision about proposal has been made positively following one of two rules mentioned above,
-then proposal becomes **approved**.
+If a decision about proposal has been made positively following one of two rules [mentioned above](#poll-and-decision-agreement-rules),
+the proposal becomes **approved**.
+
```
[the remark this answers is not in the document]
Comment (georgeee) [reply docs/block-processing/us.md]: Fixed

File: docs/block-processing/us.md [hunk 9]
```diff
@@ -72,49 +106,66 @@ and it has more stakes "for" than "against", then the proposal becomes approved
...
-Update proposal may be in five states:
+TODO: add diagram as in [section](#block-version-life-cycle)
+
+There are five possible states for Update proposal:
+
 * **Active**  
 When an update proposal gets into the blockchain within some block, it becomes _active_.
-This state means that the proposal is known in the blockchain but 
-it wasn't still decided by stakeholders whether to approve or to reject this proposal.
+
```
[the remark this answers is not in the document]
Comment (georgeee) [reply docs/block-processing/us.md]: Fixed

File: docs/block-processing/us.md [hunk 10]
```diff
@@ -72,49 +106,66 @@ and it has more stakes "for" than "against", then the proposal becomes approved
...
-If a proposal has been _approved_ in some block and there are at least `k` blocks after this one, then
+If a proposal has been _approved_ in some block and there are at least `k` blocks after this one, the
 update proposal becomes **confirmed**.
+
 _Confirmed_ state reflects the fact that _approved_ state cannot be changed anymore 
 because we have guarantee that at most `k` blocks may be rolled back.
 
 * **Discarded**  
-If a proposal has been _rejected_ in some block and there are at least `k` blocks after this one, then
+If a proposal has been _rejected_ in some block and there are at least `k` blocks after this one, the
```
[the remark this answers is not in the document]
Comment (georgeee) [reply docs/block-processing/us.md]: See above

File: docs/block-processing/us.md [hunk 11]
```diff
@@ -72,49 +106,66 @@ and it has more stakes "for" than "against", then the proposal becomes approved
...
 
-If a developer wants to change some protocol rules, for example, `Address` format, 
-he must propose update with new `BlockVersion`
+[BIP-99](https://github.com/bitcoin/bips/blob/ed283b05b332b85b6fd683be3a5d73fab6c15554/bip-0099.mediawiki) provides excellent criteria to distinguish between these two fork types:
+
+* A **soft fork** introduces new rules, or restrictions, on blocks. That way, everything that was previously invalid remains invalid, while some blocks that would have been previously considered valid become invalid.
+* A **hard fork** is a fork that makes previously invalid blocks valid.
+Soft forks have some deployment advantages like backward compatibility, and they don’t require everyone’s consensus, as the stake majority of users can impose the new rules. By contrast, hard forks require all users to upgrade.
+
+In theory, a hard fork may lead to a situation when a network splits into two parts, each maintaining a separate chain: one from the nodes that adopted the latest system update, and another from the nodes that rejected to do that. This means some blocks from the first part are considered invalid by the other part, and vice versa.
```
[the remark this answers is not in the document]
Comment (georgeee) [reply docs/block-processing/us.md]: Rewritten

File: docs/block-processing/us.md [hunk 12]
```diff
@@ -124,12 +175,31 @@ data BlockVersion = BlockVersion
...
+   * It allows independent developers to introduce multiple changes to the protocol simultaneously. For example, if one vendor decides to introduce feature X via a soft fork, and another proposes feature Y (also via a soft fork), their software will be issuing blocks with versions a.b.X and a.b.Y, which can coexist in the blockchain (see [section](#competing-block-version)). However, only one will eventually be adopted.
+
+Block version is also often referred to as **protocol version**.
+
+### Protocol constants
+
+Cardano SL supports number of constants associated with block version.
+
+Protocol constants provide us with more freedom to express protocol updates via softforks (i.e. without need for hardforks).
+Clients which run old version of protocol will be able to follow and validate blockchain without being update if for instance:
```
[the remark this answers is not in the document]
Comment (georgeee) [reply docs/block-processing/us.md]: Rewrote the sentence

File: docs/block-processing/us.md (hunk 2 above)
Comment (georgeee) [reply docs/block-processing/us.md]: As we agreed later, it's right @pva701

**Comment by pva701** [comment]
Will you squash commits?

**Comment by pva701** [comment]
Did you forget to fix `docs/block-processing/us-bv-states.png`?

we still can have such situation
```
set of competing versions {1.1.0, 2.0.0}
last adopted = 1.0.0```
 and `1.1.0` becomes adopted first and `2.0.0` next

**Pull request merged**